ui-validation/src/main/kotlin/edu/stanford/spezi/ui/validation/Validate.kt

```diff
@@ -37,18 +41,28 @@ fun Validate(
 fun Validate(
     input: String,
     rules: List<ValidationRule>,
-    modifier: Modifier = Modifier,
-    validationDebounce: Duration = DEFAULT_VALIDATION_DEBOUNCE_DURATION,
+    coroutineScope: CoroutineScope = rememberCoroutineScope(),
+    debounceDuration: Duration = DEFAULT_VALIDATION_DEBOUNCE_DURATION,
     content: @Composable () -> Unit,
 ) {
-    val validationEngineConfiguration = LocalValidationEngineConfiguration.current
+    val configuration = LocalValidationEngineConfiguration.current
     val engine = remember {
         ValidationEngineImpl(
             rules,
-            validationDebounce,
-            validationEngineConfiguration
+            configuration,
+            coroutineScope,
+            debounceDuration,
         )
     }
+    LaunchedEffect(configuration) {
```

Comment:
Launching coroutines in compose is expensive, and you are currently allocating and launching three. we can reduce into one LaunchedEffect(config, debounce, scope)

Reply:
I'm fine with combining them